vortex-array/src/opaque.rs

```diff
@@ -0,0 +1,37 @@
+use std::fmt::Debug;
+
+use vortex_error::{vortex_bail, VortexResult};
+
+use crate::encoding::{ArrayEncoding, EncodingId};
+use crate::{Array, ArrayTrait, Canonical};
+
+/// An encoding of an array that we cannot interpret.
+///
+/// Vortex allows for pluggable encodings. This can lead to issues when one process produces a file
+/// using a custom encoding, and then another process without knowledge of the encoding attempts
+/// to read it.
+///
+/// `OpaqueEncoding` allows deserializing these arrays. Many common operations will fail, but it
+/// allows deserialization and introspection in a type-erased manner on the children and metadata.
+#[derive(Debug, Clone, Copy)]
+pub struct OpaqueEncoding;
+
+pub const OPAQUE_ENCODING_ID: EncodingId = EncodingId::new("vortex.opaque", 0u16);
```

Comment:
put this id in `encoding.rs` with the other IDs, just rename `RESERVED = 0` to `OPAQUE_ENCODING_ID`

Reply:
See latest commit, I actually think it's better to propagate the original encoding ID for two reasons

1. Error messages are only helpful if you have a way to find the original ID
2. You can in theory just send an array with OpaqueEncoding back out over the wire without losing information